docs/templates.md

```diff
@@ -0,0 +1,175 @@
+# Templates
+
+Sydent uses parametrised templates to generate the content of emails it
+sends and webpages it shows to users. Example templates can be found in
+the [res](https://github.com/matrix-org/sydent/tree/main/res) folder.
+
+## Branding
+
+Sending a value for `brand` to some API endpoints allows for different
+email and http templates to be used. These templates should be stored
+in a file structure like this:
+
+```
+ root_template_dir/
+     brand1/
+         [ brand1 template files ]
+     brand2/
+         [ brand2 template files ]
+```
+
+The config option `templates.root_directory` should be set to the path
+of `root_template_dir` and `templates.default_brand` should be set to
+the sub directory to use if no `brand` (or an invalid `brand`) is provided
+by the request.
+
+Here are the requests that can contain a value for `brand` and what
+template the brand value is used to select:
+
+Invite email templates:
+ - `POST /_matrix/identity/v2/validate/msisdn/requestToken`
+
+Verification email templates:
+ - `POST /_matrix/identity/v2/validate/email/requestToken`
+
+Verification SMS templates - BRAND CURRENTLY IGNORED:
+ - `POST /_matrix/identity/v2/validate/msisdn/requestToken`
+
+Verification response templates:
+ - `GET /_matrix/identity/v2/validate/email/submitToken`
+ - `GET /_matrix/identity/v2/validate/msisdn/submitToken`
+
+
+## Template formats
+
+### `.eml` template files
+
+Files ending in ".eml" can use python `%(variable)s` string substitution.
+
+Appending "_forurl" or "_forhtml" to any of the variable names listed below
+returns their values encoded suitably for URLs or HTML respectively.
+
+e.g. ">" in the HTML encoded version would be replaced with "&gt".
```

Comment:
I'm not even sure if we should be documenting how to do this. We should really encourage people to use the Jinja templates.

Reply:
Should I put a deprecated warning around them or just get rid of them?

If someone was trying to migrate from the `.eml` files to `.eml.j2` ones then it might be helpful to have documentation on how they work

Reply:
Yeah, let's use this as an opportunity to officially deprecate them. (We'll need to ensure this goes in the changelog too.)

Reply:
Personally I feel we don't need to document the old-school templates, we're hoping to see them gone soon

Reply:
I've put a deprecated notice next to the .eml information and updated the changelog, we can always remove the .eml documentation when we drop support?